ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerPlaintextTest.java

```diff
@@ -0,0 +1,157 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static com.github.ambry.server.RouterServerTestFramework.getRouterProperties;
+
+
+public class RouterServerPlaintextTest {
+  private static MockCluster plaintextCluster;
+  private static RouterServerTestFramework testFramework;
+
+  @BeforeClass
+  public static void initializeTests()
+      throws Exception {
+    MockNotificationSystem notificationSystem = new MockNotificationSystem(9);
+    plaintextCluster = new MockCluster(notificationSystem, false, SystemTime.getInstance());
+    plaintextCluster.startServers();
+    testFramework = new RouterServerTestFramework(getRouterProperties("DC1"), plaintextCluster, notificationSystem);
+  }
+
+  @AfterClass
+  public static void cleanup()
+      throws IOException {
+    testFramework.cleanup();
+    long start = System.currentTimeMillis();
+    System.out.println("About to invoke cluster.cleanup()");
+    if (plaintextCluster != null) {
+      plaintextCluster.cleanup();
+    }
+    System.out.println("cluster.cleanup() took " + (System.currentTimeMillis() - start) + " ms.");
+  }
+
+  @Test
+  public void interleavedOperationsTest()
+      throws Exception {
+    List<OperationChain> opChains = new ArrayList<>();
+    for (int i = 0; i < 20; i++) {
+      Queue<OperationType> operations = new LinkedList<>();
+      switch (i % 3) {
+        case 0:
+          operations.add(OperationType.PUT_NB);
+          operations.add(OperationType.AWAIT_CREATION);
+          operations.add(OperationType.GET_NB);
+          operations.add(OperationType.GET_INFO_NB);
+          operations.add(OperationType.DELETE_NB);
+          operations.add(OperationType.AWAIT_DELETION);
+          operations.add(OperationType.GET_DELETED_NB);
+          operations.add(OperationType.GET_INFO_DELETED_NB);
+          break;
+        case 1:
+          operations.add(OperationType.PUT_NB);
+          operations.add(OperationType.AWAIT_CREATION);
+          operations.add(OperationType.DELETE_NB);
+          operations.add(OperationType.AWAIT_DELETION);
+          operations.add(OperationType.GET_DELETED_NB);
+          operations.add(OperationType.GET_INFO_DELETED_NB);
+          operations.add(OperationType.GET_DELETED_NB);
+          operations.add(OperationType.GET_INFO_DELETED_NB);
+          break;
+        case 2:
+          operations.add(OperationType.PUT_NB);
+          operations.add(OperationType.AWAIT_CREATION);
+          operations.add(OperationType.GET_NB);
+          operations.add(OperationType.GET_NB);
+          operations.add(OperationType.GET_NB);
+          operations.add(OperationType.GET_INFO_NB);
+          break;
+      }
+      opChains.add(testFramework.startOperationChain(32 * 1024, i, operations));
```

Comment:
I think this value should be defined as a constant (maybe even randomize it between runs?).